app/Http/Controllers/BillController.php

```diff
@@ -94,4 +94,28 @@ public function show($id)
                            ->withErrors(trans('bills.common.error_message'));
         }
     }
+    /**
+     * Destroy the specified bill from database.
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $errors = trans('bills.delete.error_message');
+        try {
+            $bill = Bill::findOrFail($id);
+            $billDetails = $bill->billDetails;
+            foreach ($billDetails as $billDetail) {
```

Comment:
[rank A] please investigate about transaction in Laravel and batch delete. We should avoid use a loop to execute many queries.